pytest_fixtures/component/katello_agent.py

```diff
@@ -0,0 +1,55 @@
+import pytest
+
+from robottelo.config import settings
+from robottelo.utils.installer import InstallerCommand
+
+
+@pytest.fixture(scope='module')
+def sat_with_katello_agent(module_target_sat):
+    """Enable katello-agent on module_target_sat"""
+    module_target_sat.register_to_cdn()
+    # Enable katello-agent from satellite-installer
+    result = module_target_sat.install(
+        InstallerCommand('foreman-proxy-content-enable-katello-agent true')
+    )
+    assert result.status == 0
+    # Verify katello-agent is enabled
+    result = module_target_sat.install(
+        InstallerCommand(help='| grep foreman-proxy-content-enable-katello-agent')
+    )
+    assert 'true' in result.stdout
+    yield module_target_sat
+
+
+@pytest.fixture
+def katello_agent_client_for_upgrade(sat_with_katello_agent, rhel_upgrade_content_host, module_org):
+    """Register content host to Satellite and install katello-agent on the host."""
+    client_repo = settings.repos['SATCLIENT_REPO'][
+        f'RHEL{rhel_upgrade_content_host.os_version.major}'
+    ]
+    sat_with_katello_agent.cli.SimpleContentAccess.disable(
+        {'organization-id': module_org.id}
+    )  # SCA enabled auto
```

Comment:
Since this is modifying the Satellite, would it make more sense to include this in the `sat_with_katello_agent` fixture itself?

Reply:
The `sat_with_katello_agent` fixture is used across other non-upgrade scenario tests as well. Hence it could have adverse effect on the other tests.

Reply:
I removed the disabling SCA, not required now.